workflow/common/util.go

```diff
@@ -325,6 +325,12 @@ func substituteParams(tmpl *wfv1.Template, globalParams, localParams map[string]
 		}
 		replaceMap["inputs.parameters."+inParam.Name] = *inParam.Value
 	}
+	//allow {{inputs.parameters}} to fetch the entire input parameters list as JSON
+	jsonInputParametersBytes, err := json.Marshal(globalReplacedTmpl.Inputs.Parameters)
+	if err != nil {
+		return nil, errors.InternalWrapError(err)
+	}
+	replaceMap["inputs.parameters"] = string(jsonInputParametersBytes)
```

Comment:
Should we limit the string length,  I'm afraid the whole parameters will be too long for env

Reply:
Usually all parameters end up in the command-line anyways (usually prefixed by parameter flags). Passing them as a single argument won't significantly. change the command-line length.